### PR TITLE
Reproducible RPMs

### LINKED ISSUE #18: RPM builds are not reproducible
RPMs build by the packager are not reproducible.

Following issues have been found:
- Dependency sorting is not stable, if two dependencies are equal in name and version, but differ in their flags, they are sorted differently each time, leading to a different RPM
- ~~Payload order is not stable~~ Caused by the rpm-builder maven plugin

~~There is probably more, will add it to the issue, as soon as I find it.~~ Seems to be the only issue, the rest seems to be caused by the rpm-builder maven plugin

### PR DESCRIPTION
**Pull request by tisoft**
Fixes #18

### CODE REVIEW

File: rpm/src/main/java/org/eclipse/packager/rpm/deps/Dependencies.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ private static void putDependencies ( final Header<RpmTag> header, final Collect
         // first sort
 
         final List<Dependency> deps = new ArrayList<> ( dependencies );
-        Collections.sort ( deps, comparing ( Dependency::getName ).thenComparing ( comparing ( Dependency::getVersion, nullsFirst ( naturalOrder () ) ) ) );
+        Collections.sort ( deps, comparing ( Dependency::getName ).thenComparing ( comparing ( Dependency::getVersion, nullsFirst ( naturalOrder () ) ) ).thenComparing ( comparing ( d -> RpmDependencyFlags.encode ( d.getFlags () ), naturalOrder () ) ) );
```
Comment (dwalluck) [inline rpm/src/main/java/org/eclipse/packager/rpm/deps/Dependencies.java:88]: If we made `Dependency` implement `Comparable`, then we could just call `Collections.sort()`.

**Comment by ctron** [comment]
This looks great. And thank you @dwalluck for taking a look at this.

I think the copyright headers need updating. But other than that, I think this can be merged.

**Comment by tisoft** [comment]
@ctron Updated the headers.

**Review by ctron — APPROVED** [review]

**Comment by ctron** [comment]
@dwalluck are you ok with this? If so, feel free to merge.

**Comment by dwalluck** [comment]
> @dwalluck are you ok with this? If so, feel free to merge.

@ctron Looks good to me. Which of the three commit options are we using---it looks like "Rebase and merge"?

**Comment by ctron** [comment]
> > @dwalluck are you ok with this? If so, feel free to merge.
> 
> @ctron Looks good to me. Which of the three commit options are we using---it looks like "Rebase and merge"?

Whatever you think makes sense. If keeping separate comments feels good, then keep the history. If is just feels like spamming the log, merge/squash it.

**Comment by tisoft** [comment]
Is there anything I can do to get this merged? 😀

**Pull request merged**